SystemAudioPlayer/SystemAudioPlayerImplementation.cpp

```diff
@@ -199,18 +210,20 @@ namespace Plugin {
         _adminLock.Lock();
          player = getObjectFromMap(id); 
         _adminLock.Unlock();
-        if(player != NULL)
-        {
-            if(player->getSourceType() == SourceType::FILESRC)
-            {
-                if(!extractFileProtocol(url))
-                {
+
+         if(player != NULL) {
+            string sourceType = sourceTypeToString(player->getSourceType());
+            if(!std::regex_match(url, patternMap.at(sourceType))) {
+                SAPLOG_ERROR("SAP: SystemAudioPlayerImplementation Source %s and Url %s is different",sourceType.c_str(),url.c_str());
+                returnResponse(false);
+            }
+            if(sourceType == "filesrc") {
```

Comment:
Please have a new line before the sourceType check at 220

Reply:
done